src/extractor/guidance/roundabout_handler.cpp

```diff
@@ -329,23 +326,13 @@ RoundaboutType RoundaboutHandler::getRoundaboutType(const NodeID nid) const
         if (1 == roundabout_name_ids.size() &&
             0 == connected_names.count(*roundabout_name_ids.begin()))
             return RoundaboutType::Rotary;
-        else
-            return RoundaboutType::Roundabout;
     }
 
-    if (radius <= MAX_ROUNDABOUT_INTERSECTION_RADIUS)
```

Comment:
I find it highly dangerous not to have a radius check in here. There are roundabouts of more than 2km in range (special case, though real). Announcing all of them as turns is very dangerous.

Also, if we have a long distance traveled in the roundabout, it is likely that `straight` is not obvious anymore.